pkg/flow/adapter/xslttransform/adapter_test.go

```diff
@@ -256,6 +258,45 @@ func TestXSLTTransformEvents(t *testing.T) {
 	}
 }
 
+func TestXSLTTransformToSink(t *testing.T) {
+	testCases := map[string]struct {
+		xslt          string
+		inEvent       cloudevents.Event
+		expectedEvent cloudevents.Event
+	}{
+		"transform ok": {
+			xslt:          tXSLT,
+			inEvent:       newCloudEvent(tXML, cloudevents.ApplicationXML),
+			expectedEvent: newCloudEvent(tOutXML, cloudevents.ApplicationXML),
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			ceClient := adaptertest.NewTestClient()
+			ctx := context.Background()
+			style, err := parseXSLT(tc.xslt)
+			assert.NoError(t, err)
+
+			a := &xsltTransformAdapter{
+				logger: logtesting.TestLogger(t),
+
+				ceClient:     ceClient,
+				xsltOverride: false,
+				defaultXSLT:  style,
+				sink:         "http://localhost:8080",
+			}
+
+			e, r := a.dispatch(ctx, tc.inEvent)
+			assert.Nil(t, e)
+			assert.Equal(t, cloudevents.ResultACK, r)
+
+			events := ceClient.Sent()
+			require.Equal(t, 1, len(events))
+			assert.Equal(t, tc.expectedEvent, events[0])
```

Comment:
The test is failing here.

Both the data and context attributes seem to match, but I'm not positive you can just compare deep structs using the `assert` module, due to nested pointers, private fields, etc.

You're going to have to compare attributes and data separately. Example (without data comparison, but you can compare data the same way):
https://github.com/triggermesh/triggermesh/blob/9376615c7461c715eb7578d40ad1814324e8e7e5/pkg/sources/adapter/awssqssource/adapter_test.go#L163-L169

Reply:
cant i just do this? https://github.com/triggermesh/triggermesh/pull/503/commits/071681fcf7980174545706149f93df0eb3c5ea6d